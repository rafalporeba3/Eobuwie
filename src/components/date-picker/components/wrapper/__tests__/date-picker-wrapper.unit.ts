import { createLocalVue, Wrapper }   from '@vue/test-utils';
import Vuex                          from 'vuex';
import CompositionApi                from '@vue/composition-api';
import { createTestInstanceShallow } from '@/__tests__/factory';
import cDatePickerWrapper            from '../date-picker-wrapper.component.vue';

const localVue = createLocalVue();
localVue.use(CompositionApi);
localVue.use(Vuex);

describe('Date picker wrapper component', () => {
  let wrapper!: Wrapper<any>;

  beforeEach(() => {
    const { wrapperInstance } = createTestInstanceShallow(cDatePickerWrapper, {
      localVue,
      propsData: {
        title: 'Test',
      },
      slots: {
        default: '<span>Test slot</span>',
      },
    });

    wrapper = wrapperInstance;
  });

  it('should render cDatePickerWrapper component with correct class', () => {
    expect(wrapper.classes())
      .toContain('c-date-picker-wrapper');
  });

  it('should render title element', () => {
    const titleElement: Wrapper<any> = wrapper.find('.c-date-picker-wrapper__title');

    expect(titleElement.text())
      .toBe('Test');
  });

  it('should render slot when it has been passed', () => {
    expect(wrapper.text())
      .toContain('Test slot');
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
