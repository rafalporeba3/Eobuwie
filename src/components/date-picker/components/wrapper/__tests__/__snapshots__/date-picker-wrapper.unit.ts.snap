// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date picker wrapper component should match snapshot 1`] = `
<div class="c-date-picker-wrapper"><span class="c-date-picker-wrapper__title">

    Test
  </span> <span>Test slot</span></div>
`;
